temporalio/lib/temporalio/internal/worker/workflow_instance.rb

```diff
@@ -238,6 +243,9 @@ def activate_internal(activation)
           @commands = []
           @current_activation_error = nil
           @continue_as_new_suggested = activation.continue_as_new_suggested
+          @current_deployment_version = WorkerDeploymentVersion._from_bridge(
+            activation.deployment_version_for_current_task
+          )
```

Comment:
Negligible performance benefit, but can also do this lazily (e.g. `nil` this and `@current_deployment_version ||=` in a `def current_deployment_version` method instead of `attr_reader`) instead of creating a new, mostly-never-used object for every activation (especially one that probably doesn't change much either)

Reply:
:man_shrugging: hardly seems worth the extra code

Reply:
It's not much extra code I don't think, but I think it's a reasonable Ruby pattern to lazy create something that's rarely accessed